### PR TITLE
Removing variables and dependent variables rebased

### PR DESCRIPTION
**Pull request by hgohil2805**
- can delete all variables without the error msg.
- can delete variables if dependent variables are being deleted.
- will not allow deletion of original variables if dependent variables still exists.

### CODE REVIEW

File: vistrails/gui/uvcdat/definedVariableWidget.py [hunk 1]
```diff
@@ -449,14 +449,29 @@ def clear(self):
             d.show()
             self.ieds.append(d)
 
-    def trashVariable(self):
+    '''This method calls the delete variable method 2 times. The first time returns the list of 
+       variables which have dependent variable and therefore cannot be deleted till the dependent 
+       variables are. The second time goes through this list and tries to delete them again 
+       after (if) their dependent variables have been deleted
+    '''
```
Comment (remram44) [inline vistrails/gui/uvcdat/definedVariableWidget.py:456]: docstrings go below `def:` lines

File: vistrails/gui/uvcdat/definedVariableWidget.py [hunk 2]
```diff
@@ -261,7 +261,7 @@ def addVariable(self, var, type_='CDMS'):
         ## print "Added variable"
         ## self.emit(QtCore.SIGNAL('setupDefinedVariableAxes'), var)
 
-    def deleteVariable(self, varid):
+    def deleteVariable(self, varid,count):
```
Comment (remram44) [inline vistrails/gui/uvcdat/definedVariableWidget.py:264]: What is this `count` thing? It seems it is either 1 or 2?

Comment (hgohil2805) [inline vistrails/gui/uvcdat/definedVariableWidget.py:264]: I am keeping a count on how many times the method is being called. Depending on the first or the second call the method either returns the var or throws the error.

File: vistrails/gui/uvcdat/definedVariableWidget.py (hunk 1 above)
Comment (hgohil2805) [inline vistrails/gui/uvcdat/definedVariableWidget.py:456]: ok

**Comment by remram44** [comment]
Issue: UV-CDAT/uvcdat#897

File: vistrails/gui/uvcdat/project_controller.py [hunk 3]
```diff
@@ -87,6 +87,7 @@ def __init__(self, vt_controller, name=''):
     def add_defined_variable(self, var):
         try:
             self.defined_variables[var.name] = var
+            print 'variable name is: %s' (str (var))
```
Comment (remram44) [inline vistrails/gui/uvcdat/project_controller.py:90]: debug code?

**Comment by remram44** [comment]
:exclamation: There is no reason why trying twice should be better than once. This might require any number of calls (at most, the number of variables).

Correct fix is to toposort the variables and delete in the right order.

File: vistrails/gui/uvcdat/project_controller.py (hunk 3 above)
Comment (remram44) [inline vistrails/gui/uvcdat/project_controller.py:90]: this even makes the method crash, "str object is not callable"

**Comment by remram44** [comment]
<a href="https://cloud.githubusercontent.com/assets/426784/6416925/de23439e-be77-11e4-9814-6ac928e99f91.PNG"></a> <a href="https://cloud.githubusercontent.com/assets/426784/6416926/e06d7b24-be77-11e4-858e-7ba7b096e8a0.PNG"></a> <a href="https://cloud.githubusercontent.com/assets/426784/6416927/e28c2c20-be77-11e4-963b-0f873d2ffbc0.PNG"></a>

File: vistrails/gui/uvcdat/project_controller.py (hunk 3 above)
Comment (hgohil2805) [inline vistrails/gui/uvcdat/project_controller.py:90]: nope. That's code that already exists. Not something I wrote.

Comment (remram44) [inline vistrails/gui/uvcdat/project_controller.py:90]: We can all see the diff here. You definitely wrote this.

**Comment by hgohil2805** [comment]
Yes. unfortunately the method I wrote yesterday only works for 2 levels of variable dependencies. 

I am trying to fix my build and get a fresh one today on a new VMware OS. Will complete this issue once I have a good build to work on.

File: vistrails/gui/uvcdat/project_controller.py (hunk 3 above)
Comment (hgohil2805) [inline vistrails/gui/uvcdat/project_controller.py:90]: Oh yes the print line is something I was using. my bad .

Comment (hgohil2805) [inline vistrails/gui/uvcdat/project_controller.py:90]: will take care of it for the next PR

**Pull request closed without merging**